tck/src/main/java/jakarta/json/bind/tck/defaultmapping/polymorphictypes/AnnotationPolymorphismArrayTest.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package jakarta.json.bind.tck.defaultmapping.polymorphictypes;
+
+import java.lang.invoke.MethodHandles;
+import java.time.LocalDate;
+
+import jakarta.json.bind.Jsonb;
+import jakarta.json.bind.JsonbBuilder;
+import jakarta.json.bind.JsonbException;
+import jakarta.json.bind.annotation.JsonbCreator;
+import jakarta.json.bind.annotation.JsonbDateFormat;
+import jakarta.json.bind.annotation.JsonbPolymorphicType;
+import jakarta.json.bind.annotation.JsonbProperty;
+import jakarta.json.bind.annotation.JsonbSubtype;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.fail;
+
+/**
+ * Tests for verification of proper polymorphism handling based on annotation with array wrapping format.
+ */
+@RunWith(Arquillian.class)
+public class AnnotationPolymorphismArrayTest {
+
+    private final Jsonb jsonb = JsonbBuilder.create();
+
+    @Deployment
+    public static WebArchive createTestArchive() {
+        return ShrinkWrap.create(WebArchive.class)
+                .addPackages(true, MethodHandles.lookup().lookupClass().getPackage().getName());
```

Comment:
AnnotationPolymorphismArrayTest.class.getPackage().getName()?

Reply:
`MethodHandles.lookup().lookupClass().getPackage().getName()` does the same thing, and one can copy-paste it simply to another test without the need to change anything.

Reply:
It looks the same but `jdk.internal.reflect.Reflection#getCallerClass` would do the same and it is generally speaking pointless to do. On the copy point, any modern IDE can change the ref but if you care use a caller extractor logic instead of doing `newWrapper().unwrap()` which is what the method handles does right now - which is why it looked weird and I asked about it.

side note: if you really want to propagate this pattern use an abstract class or an arquillian extension which is the proper way to reuse the packaging rather than an inline workaround.

hope it makes sense.